packages/server/utils/OpenAIServerManager.ts

```diff
@@ -42,51 +43,78 @@ class OpenAIServerManager {
     }
   }
 
+  // setting the minimal max_tokens limit increases the speed of the request and reduces the tokens used
+  estimateOutputTokens(reflectionsText: string[]) {
+    const avgGroupNameLength = 10 // assume an average group name length of 10 tokens
+    const totalLengthInTokens = reflectionsText.reduce(
+      (total, reflectionText) => total + Math.ceil(reflectionText.length / AVG_CHARS_PER_TOKEN),
+      0
+    )
+    const avgReflectionLength = totalLengthInTokens / reflectionsText.length
+
+    const avgReflectionsPerGroup = 3 // assume each group will have around 3 reflections
+    const numGroups = Math.ceil(reflectionsText.length / avgReflectionsPerGroup)
+
+    const outputTokens = Math.round(
+      numGroups * (avgGroupNameLength + avgReflectionsPerGroup * avgReflectionLength)
```

Comment:
+1 isn't this equal to `numGroups * avgGroupNameLength + totalLengthInTokens`?

Reply:
Good point, updated!